src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+/* eslint-disable @typescript-eslint/lines-between-class-members */
+export type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: string;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape = 'triangle';
+  color: Color;
+  private a: number;
+  private b: number;
+  private c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    this.color = color;
+
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    const maxSide = Math.max(a, b, c);
+    const sumOfSides = a + b + c;
+
+    if (maxSide >= sumOfSides - maxSide) {
+      throw new Error('This triangle cannot exist');
+    }
+
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
```

Comment:
u can use next syntax in all ur cases. but it's just another way
```suggestion
  constructor(
   public color: Color, 
   public a: number,
   public b: number,
   public c: number
  ) {
    if (a <= 0 || b <= 0 || c <= 0) {
      throw new Error('Side lengths must be greater than 0');
    }

    const maxSide = Math.max(a, b, c);
    const sumOfSides = a + b + c;

    if (maxSide >= sumOfSides - maxSide) {
      throw new Error('This triangle cannot exist');
    }
  }
```